Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.h

```diff
@@ -44,10 +44,10 @@ class RemoteRenderingBackend;
 class RemoteImageBufferProxy : public WebCore::ImageBuffer {
 public:
     template<typename BackendType>
-    static RefPtr<RemoteImageBufferProxy> create(const WebCore::FloatSize& size, float resolutionScale, const WebCore::DestinationColorSpace& colorSpace, WebCore::PixelFormat pixelFormat, WebCore::RenderingPurpose purpose, RemoteRenderingBackendProxy& remoteRenderingBackendProxy)
+    static RefPtr<RemoteImageBufferProxy> create(const WebCore::FloatSize& size, float resolutionScale, const WebCore::DestinationColorSpace& colorSpace, WebCore::PixelFormat pixelFormat, WebCore::RenderingPurpose purpose, RemoteRenderingBackendProxy& remoteRenderingBackendProxy, RemoteRenderingBackendProxy::AvoidIOSurfaceSizeCheckInWebProcess avoidIOSurfaceSizeCheckInWebProcess)
     {
         auto parameters = WebCore::ImageBufferBackend::Parameters { size, resolutionScale, colorSpace, pixelFormat, purpose };
-        if (BackendType::calculateSafeBackendSize(parameters).isEmpty())
+        if (BackendType::calculateSafeBackendSize(parameters).isEmpty() && avoidIOSurfaceSizeCheckInWebProcess == RemoteRenderingBackendProxy::AvoidIOSurfaceSizeCheckInWebProcess::No)
```

Comment:
I think the order should be reversed:
`
if (avoidIOSurfaceSizeCheckInWebProcess == RemoteRenderingBackendProxy::AvoidIOSurfaceSizeCheckInWebProcess::No && BackendType::calculateSafeBackendSize(parameters).isEmpty())
`